src/AutoRest.CSharp/Common/Output/Models/Types/SignatureType.cs

```diff
@@ -0,0 +1,318 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Common.Output.Expressions.Statements;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Common.Output.Models;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.Output.Models.Shared;
+using Microsoft.CodeAnalysis;
+using static AutoRest.CSharp.Common.Output.Models.Snippets;
+
+namespace AutoRest.CSharp.Output.Models.Types
+{
+    /// <summary>
+    /// This type holds three portions of codes:
+    ///     - current
+    ///     - custom
+    ///     - baseline contract
+    ///     current union custom compare with baseline contract outputs the changeset, we can apply different rules with it.
+    /// </summary>
+    internal class SignatureType
+    {
+        private readonly TypeFactory _typeFactory;
+        private readonly string _defaultNamespace;
+        private readonly string _defaultName;
```

Comment:
we should simply have:
```suggestion
        private readonly string _namespace;
        private readonly string _name;
```